Create user.ts

pranavgowrish opened the pull request:
close #1999

pranavgowrish committed:
Create user.ts

pranavgowrish committed:
Update user.ts

pranavgowrish committed:
Update user.ts

pranavgowrish committed:
Update user.ts

khandrew1 commented on src/schemas/user.ts:
```diff
@@ -0,0 +1,38 @@
+  MAJORS,
+  AGES,
+  DIETS,
+} from "@/data/form/information"; // Common imports found in most schemas
```
remove comments

khandrew1 commented on src/schemas/user.ts:
```diff
@@ -0,0 +1,38 @@
+  DIETS,
+} from "@/data/form/information"; // Common imports found in most schemas
+
+// following exports are common in 9 out of 12 schemas
```
what about the other 3? what are they and what are you doing to handle them.

pranavgowrish replied:
This user.ts can be imported in the following schemas for common fields:

1. admin.ts
2.  committee.ts
3. judge.ts
4. lead.ts
5. mentor.ts
6. participant.ts
7. volunteer.ts

For the rest, the fields are few so there's little common:

1. feedback.ts
2. idea.ts
3. interest.ts
4. panel.ts
5. sponsor.ts

pranavgowrish committed:
Update user.ts

This user.ts can be imported in the following schemas for common fields:

1. admin.ts
2.  committee.ts
3. judge.ts
4. lead.ts
5. mentor.ts
6. participant.ts
7. volunteer.ts

For the rest, the fields are few so there's little common:

1. feedback.ts
2. idea.ts
3. interest.ts
4. panel.ts
5. sponsor.ts

pranavgowrish committed:
Update user.ts

shahdivyank commented:
once the user.ts file is made, the other forms should be inheriting from it as well
